### PR TITLE
Add PyTorch utility, implement PyTorch VAE and AutoEncoder

### PR DESCRIPTION
**Pull request by RaymondY**
### All Submissions Basics:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked all [Issues](../../issues) to tie the PR to a specific one?

### All Submissions Cores:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### New Model Submissions:

* [x] Have you created a <NewModel>.py in ~/pyod/models/?
* [x] Have you created a <NewModel>_example.py in ~/examples/?
* [x] Have you created a test_<NewModel>.py in ~/pyod/test/?
* [x] Have you lint your code locally prior to submission?

### CODE REVIEW

**Comment by yzhao062** [comment]
it appears that mac tests failed -- maybe take a look but I could also help!

**Comment by RaymondY** [comment]
`torch_utility` includes:
* class `PyODDataset`, custom PyTorch dataset
* class `LinearBlock`, includes linear layer, batch normalization, activation function, dropout layer, and linear layer weight init
* func `get_activation_by_name`, includes 7 common activation functions
* func `init_weights`, includes all weight initialization methods

Utility functions and classes are applied in VAE and AutoEncoder.
